util/cloud/aws.h

```diff
@@ -43,6 +44,7 @@ class AWS {
   void SetScopeAndSignKey();
 
   std::string region_, service_, secret_, access_key_;
+  std::optional<std::string> session_token_;
```

Comment:
you could get away with using an empty value instead of using `optional`. Why do you prefer optional? readability?

Reply:
Yes, empty value will work for sure. Optional just clearly shows this is an optional value, feels somewhat cleaner. But I can change it back